src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -344,6 +381,31 @@ protected final void handleSelectedKey(final SelectionKey sk) throws IOException
 
   }
 
+  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
+  {
+    final Thread currentThread = Thread.currentThread();
+    if (eventThread == currentThread) {
+      int size = tasks.size();
+      if (size == tasks.capacity()) {
```

Comment:
These kind of checks cause code bloat and also forces one to understand the logic and infer the intentions. Best to avoid unnecessary checks. Assert is a middle ground where in one line you convey that it's a logic error if assert fails. Technically even assert  should not exist in non-debug builds - some programming languages (e.g. C/c++) handle that well.